third-culture-software/bhima
# Automatically merge lots for inventory items with zero quantity in stock

This PR adds a command to the main [Menu] under the "Stock > Find Duplicate Lots" page that allows automatic merging of all lots for (specific inventory items) with zero quantity in stock.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5740

**TESTING**
- Use a production database
- Make sure the 1.19 - 1.20 SQL changes are executed
- Go to Stock > Find Duplicate Lots
- Try the new function (in the top [Menu])
- It should reduce the number of duplicate lots significantly.

NOTE: After doing this, there will still be some lots with zero quantity in stock.    That is because the new function only automerges lots with zero quantity in stock over all lots for an inventory type.   Whereas the main display does it for combinations of inventory type AND lot labels.  

ALSO: This PR fixes a bug in the quantity-in-stock for the main display; it now should show the correct result (before it was showing the quantity in stock for whichever item the "GROUP BY" selected.
